### PR TITLE
Cart page

### PR DESCRIPTION
**Pull request by agrawalshanu708**
design cart page
Design wishlist page
-Added functionalities

### CODE REVIEW

File: src/backend/db/products.js [hunk 1]
```diff
@@ -64,7 +65,8 @@ import { v4 as uuid } from "uuid";
                     "price":3699,
```
Comment (zabihshaik) [inline src/backend/db/products.js:65]: Better to use Prettier for the code to look good and formatted.

File: src/components/product-gallery/cardView.jsx [hunk 2]
```diff
@@ -1,12 +1,71 @@
...
+ <BsCartPlus onClick = {() => cartDispatch({
+   type:"ADD_TO_CART",
+   payload: {_id,tittle,description,price,category,categoryName,qty,new_arrival,original_price,discount,image,rating:{rate,count}}
+    })} size="3rem"/>
+ </div>
+ </div>
+ </div>
+   )}
+    
+
```
Comment (zabihshaik) [inline src/components/product-gallery/cardView.jsx:54]: Better to remove unnecessary blank spaces.

File: src/components/product-gallery/cardView.jsx [hunk 3]
```diff
@@ -1,12 +1,71 @@
...
+   )}
+    
+  
+
+
+ 
+
+
+  
+  export{CardView}
```
Comment (harshitpaliwal95) [inline src/components/product-gallery/cardView.jsx:61]: remove empty space

**Review by harshitpaliwal95 — COMMENTED** [review]
suggestion 
- cart page ui can be improve by replace border with box shadow
- give padding to icons in cart
over all looks good you can merge

**Review by zabihshaik — APPROVED** [review]
Everything looks good. Just a few suggestions:
1. Better to remove unnecessary blank spaces.
2. Better to use Prettier for the code to look good and formatted.
3. Better to have background for cards in Products page for differentiation.

File: src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -21,10 +24,11 @@ const  Navbar = () => {
                <span class="material-icons search-icon">search</span>
                <span class="material-icons mob-search-icon">search</span>
                <Link to = "/cart" class="cart">
-               <FiShoppingCart size="3rem"/>
+               <FiShoppingCart size="3rem"/>{cartState.itemsInCart.length}
                </Link>
                <Link to = "/wishlist"class="wishlist">
-                <BsHeart size = "3rem"/> 
+                <BsHeart size = "3rem"/>
+                {/* {wishlistState.itemsInWishlist.length} */}
```
Comment (Chiragmodi01) [inline src/components/Navbar/Navbar.jsx:31]: try not to leave commented code here and there.

File: src/pages/cart/cart.css [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    text-decoration: line-through;
+
+}
+.cart-discount{
+    color: red;
+    font-size: 1.5rem;
+}
+.qty-btn{
+    width: 2.5rem;
+    border: 1px solid #090a0c;
```
Comment (Chiragmodi01) [inline src/pages/cart/cart.css:53]: If you're using some color many times, you can store it in variable.

File: src/pages/cart/cart.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  const getDiscount = () =>   cartState.itemsInCart.reduce(getOriginalPrice,0) - cartState.itemsInCart.reduce(getTotalPrice,0)
+  const getNetPrice = () => cartState.itemsInCart.reduce(getTotalPrice,0) + getConviencefee()
+  return (
+    <>
+    <div className="cart-page">
+    <div className="outer-flex-horizontal col-6">{
+         cartState.itemsInCart.map(HorizontalCardView)
+         }</div>
+{/* ------------------------------------------ */}
```
Comment (Chiragmodi01) [inline src/pages/cart/cart.jsx:23]: not necessarily to put this in jsx

**Review by Chiragmodi01 — COMMENTED** [review]
- Quantity should not be in negative, so you can stop that at qty = 1 in '-' button in cart.
Overall UI and code structure looks good to me.

**Pull request merged**